ledger/ledger-api-common/src/main/scala/com/digitalasset/ledger/api/validation/TransactionServiceRequestValidator.scala

```diff
@@ -190,4 +192,53 @@ class TransactionServiceRequestValidator(
       }
       .fold(partyValidator.requireKnownParties(transactionFilter.filtersByParty.keys))(Left(_))
 
+  def validateEventsByContractId(
+      req: GetEventsByContractIdRequest
+  )(implicit
+      contextualizedErrorLogger: ContextualizedErrorLogger
+  ): Result[transaction.GetEventsByContractIdRequest] = {
+    for {
+      contractId <- requireContractId(req.contractId, "contract_id")
+      _ <- requireNonEmpty(req.requestingParties, "requesting_parties")
+      parties <- partyValidator.requireKnownParties(req.requestingParties)
+    } yield {
+      transaction.GetEventsByContractIdRequest(contractId, parties)
+    }
+  }
+
+  def validateEventsByContractKey(
+      req: GetEventsByContractKeyRequest
+  )(implicit
+      contextualizedErrorLogger: ContextualizedErrorLogger
+  ): Result[transaction.GetEventsByContractKeyRequest] = {
+
+    for {
+      apiContractKey <- requirePresence(req.contractKey, "contract_key")
+      contractId <- ValueValidator.validateValue(apiContractKey)
+      apiTemplateId <- requirePresence(req.templateId, "template_id")
+      templateId <- FieldValidations.validateIdentifier(apiTemplateId)
+      _ <- requireNonEmpty(req.requestingParties, "requesting_parties")
+      requestingParties <- partyValidator.requireKnownParties(req.requestingParties)
+      startExclusive <- LedgerOffsetValidator.validateOptional(
+        req.beginExclusive,
+        "start_exclusive",
+      )
+      endInclusive <- LedgerOffsetValidator.validateOptional(
+        req.endInclusive,
+        "end_inclusive",
+      )
+    } yield {
+
+      transaction.GetEventsByContractKeyRequest(
+        contractKey = contractId,
+        templateId = templateId,
+        requestingParties = requestingParties,
+        maxEvents = if (req.maxEvents != 0) req.maxEvents else 1000,
```

Comment:
We should validate above that maxEvents are non-negative. alternatively we can use uint32 in proto